tapchannelmsg/custom_channel_data.go

```diff
@@ -14,10 +14,14 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+// ChannelDataMaxRecordSize is the maximum size of the custom channel data that
+// can be read from a reader.
+const ChannelDataMaxRecordSize = tlv.MaxRecordSize * 2
```

Comment:
Have we been able to validate that this resolves the issue? 

The other thing to consider here is that w/o adding any proof level deduplication, this'll expand along with the number of active HTLCs. So perhaps it should be a multiplier based on the current max proof suffix size. We know the max amount of HTLCs in either direction, so we can base it off that. 

Also unlike other contexts, we're the one creating this value, so it's a bit more trusted.

Reply:
The data is already in memory. I've pushed up a new commit which adds a "max read size" argument which we can set based on the data that's already in memory.

I've tested this with the litd itest and it solves the max read size issue.

I've also created a unit test on this branch so that we can play around with the custom channel data which exceeded the previous max read limit:
https://github.com/lightninglabs/taproot-assets/commit/2a90bb366a0c03d68634723b6ae581e7c9266276

I didn't see anything particularly interesting in that custom channel data example. It's commitment portion just bigger than the default TLV record size.